storage/connectors/redis/src/main/java/feast/storage/connectors/redis/retriever/RedisClusterClient.java

```diff
@@ -62,18 +71,81 @@ private RedisClusterClient(Builder builder) {
     this.asyncCommands.setAutoFlushCommands(false);
   }
 
+  public static String getAddressString(String host) {
+    try {
+      return InetAddress.getByName(host).getHostAddress();
+    } catch (UnknownHostException e) {
+      throw new RuntimeException(String.format("getAllByName() failed: %s", e.getMessage()));
+    }
+  }
+
+  public static MappingSocketAddressResolver customSocketAddressResolver(
+      RedisClusterStoreConfig config) {
+
+    List<String> configuredHosts =
+        Arrays.stream(config.getConnectionString().split(","))
+            .map(
+                hostPort -> {
+                  return hostPort.trim().split(":")[0];
+                })
+            .collect(Collectors.toList());
+
+    Map<String, String> mapAddressHost =
+        configuredHosts.stream()
+            .collect(
+                Collectors.toMap(host -> ((String) getAddressString(host)), host -> (String) host));
+
+    return MappingSocketAddressResolver.create(
+        DnsResolvers.UNRESOLVED,
+        hostAndPort -> 
+            mapAddressHost.keySet().stream().anyMatch(i -> i.equals(hostAndPort.getHostText()))
+                ? hostAndPort.of(
+                    mapAddressHost.get(hostAndPort.getHostText()), hostAndPort.getPort())
+                : hostAndPort);
+  }
+
+  public static ClientResources customClientResources(RedisClusterStoreConfig config) {
+    ClientResources clientResources =
+        ClientResources.builder()
+            .nettyCustomizer(
+                new NettyCustomizer() {
+                  @Override
+                  public void afterBootstrapInitialized(Bootstrap bootstrap) {
+                    bootstrap.option(EpollChannelOption.TCP_KEEPIDLE, 15);
+                    bootstrap.option(EpollChannelOption.TCP_KEEPINTVL, 5);
+                    bootstrap.option(EpollChannelOption.TCP_KEEPCNT, 3);
+                    // Socket Timeout (milliseconds)
+                    bootstrap.option(EpollChannelOption.TCP_USER_TIMEOUT, 60000);
```

Comment:
thoughts on having these be exposed in the cluster config?

Reply:
I think that makes sense. Also, these are very cloud provider specific for Azure Cache for Redis in terms of how socket timeouts are handled (https://github.com/lettuce-io/lettuce-core/issues/1428), so I'm wondering if it makes sense to subclass `RedisClusterClient` and make it cloud-provider specific (e.g. `AzureRedisClusterClient`, `AwsRedisClusterClient`, `GcpRedisClusterClient`, etc.).

Reply:
Cloud provider can be set via config property (e.g. `cloud_provider: azure`).

Reply:
Do you have a usecase for a stateful redis connection in newer versions of feast?

Reply:
i.e. this wouldnt hold true if you wanted to run your own Redis cluster in Azure?

Might make sense then to instead allow for additional azure_configs within the RedisClusterConfig?